ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/timeline/TimelineComponentView.kt

```diff
@@ -67,11 +72,16 @@ internal fun TimelineComponentView(
     ) {
         val itemBarriers = mutableListOf<HorizontalAnchor>()
         val iconRefs = mutableListOf<ConstrainedLayoutReference>()
-        for ((index, item) in timelineState.items.withIndex()) {
-            val isLastItem = index == timelineState.items.size - 1
+        val biggestIconWidth: Dp? = remember(timelineState.items) {
+            timelineState.items
+                .maxByOrNull { it.icon.size.width.dpOrNull() ?: 0.dp }
+                ?.icon?.size?.width?.dpOrNull()
+        }
```

Comment:
- Since `timelineState.items` is a `State`, we can use `derivedStateOf` to make `biggestIconWidth` a `State` too. Could save some recompositions. 
- I think `maxOfOrNull` works here? Saves the final re-read of the `width` of the max item. 

(Don't apply cause you'll need imports)
```suggestion
        val biggestIconWidth: Dp? by remember {
            derivedStateOf {
                timelineState.items
                    .maxOfOrNull { it.icon.size.width.dpOrNull() ?: 0.dp }
            }
        }
```

Reply:
Oh you're right, thank you!!